compositor_common/src/error.rs

```diff
@@ -44,6 +44,58 @@ impl Display for UnusedNodesError {
     }
 }
 
+#[derive(Debug, thiserror::Error, PartialEq, Eq)]
+pub enum ConstraintsValidationError {
+    #[error("Specification of node \"{1}\" doesn't satisfy node input pads count constraint.")]
+    InvalidInputsPads(#[source] InputsCountValidationError, NodeId),
+}
+
+#[derive(Debug, thiserror::Error, PartialEq, Eq)]
+pub enum InputsCountValidationError {
+    #[error(
+        "Nodes that use transformation \"{}\" doesn't expect input pads, but {} input pads were specified.",
+        transformation_name,
+        defined_input_pads_count
+    )]
+    NoInputsRequired {
+        transformation_name: Arc<str>,
+        defined_input_pads_count: u32,
+    },
+    #[error(
+        "Nodes that use transformation \"{}\" expect single input pad, but {} input pads were specified.",
+        transformation_name,
+        defined_input_pads_count
+    )]
+    ExactlyOneInputRequired {
+        transformation_name: Arc<str>,
+        defined_input_pads_count: u32,
+    },
+    #[error(
+        "Nodes that use transformation \"{}\" expect {} input pads, but {} input pads were specified.",
+        transformation_name,
+        expected_input_pads_count,
+        defined_input_pads_count
+    )]
+    ExactNumberOfInputsRequired {
+        transformation_name: Arc<str>,
+        expected_input_pads_count: u32,
+        defined_input_pads_count: u32,
+    },
+    #[error(
+        "Nodes that use transformation \"{}\" expect at least {} and at most {} input pads, but {} were specified", 
+        transformation_name,
+        minimal_inputs_expected,
+        maximal_inputs_expected,
+        defined_input_pads_count
+    )]
+    InvalidBoundedInputsRequired {
+        transformation_name: Arc<str>,
+        minimal_inputs_expected: u32,
+        maximal_inputs_expected: u32,
+        defined_input_pads_count: u32,
+    },
+}
```

Comment:
@wkozyra95  
1. Do you still think we should have 4 different errors? IMO first 3 are basically the same and we could have just 2 errors defined here
2. Would you put those errors here or move them into a separate module?

Reply:
1. I don't have an opinion on that, but I want to be able to specify error messages for those cases in a more elastic way. Whether it's one error or not does not matter to me.
2. Up to you, maybe move it down a bit. (My idea is to keep top-level types higher, but this is subjective) . If error does not have any lower-level source errors then it's fine to move it somewhere else.

I raised my concerns about wording "Nodes that use transformation" before and that was subjective, but when you moved it here it's no longer the case, because now it's not correct anymore, because those errors are no longer builtin specific. Try to reword it, but if you don't have a better idea of how to phrase it then:
 - Rename errors to clarify that they are specific to built-in transformation. (and leave current wording)
 - Remove all the variants that are not currently used.
 - Add an issue to create generic error messages for other types (you can assign me to it)